packages/firestore/src/platform_node/grpc_connection.ts

```diff
@@ -135,26 +129,57 @@ export class GrpcConnection implements Connection {
         token
       };
     }
+
     return this.cachedStub.stub;
   }
 
-  private getRpc(rpcName: string, token: Token | null): any {
-    const stub = this.getStub(token);
-
+  private getRpcCallable<Req, Resp>(
+    rpcName: string
+  ): UnaryRpc<Req, Resp> | ReadableRpc<Req> | DuplexRpc {
     // RPC Methods have the first character lower-cased
     // (e.g. Listen => listen(), BatchGetDocuments => batchGetDocuments()).
     const rpcMethod = rpcName.charAt(0).toLowerCase() + rpcName.slice(1);
-    const rpc = stub[rpcMethod];
+    const rpc = this.cachedStub.stub[rpcMethod];
     assert(rpc != null, 'Unknown RPC: ' + rpcName);
+    return rpc;
+  }
+
+  private getUnaryRpc<Req, Resp>(
+    rpcName: string,
+    token: Token | null
+  ): UnaryRpc<Req, Resp> {
+    const stub = this.ensureActiveStub(token);
+    const rpc = this.getRpcCallable<Req, Resp>(rpcName) as UnaryRpc<Req, Resp>;
+    return rpc.bind(stub);
+  }
+
+  private getBidirectionalStreamingRpc<Req, Resp>(
+    rpcName: string,
+    token: Token | null
+  ): DuplexRpc {
+    const stub = this.ensureActiveStub(token);
+    const rpc = this.getRpcCallable<Req, Resp>(rpcName) as DuplexRpc;
+    return rpc.bind(stub);
+  }
 
+  private getStreamingRpc<Req, Resp>(
+    rpcName: string,
+    token: Token | null
+  ): ReadableRpc<Req> {
+    const stub = this.ensureActiveStub(token);
+    const rpc = this.getRpcCallable<Req, Resp>(rpcName) as ReadableRpc<Req>;
```

Comment:
So part of me likes these typed helper methods, but another part of me sees that:
1) They're exactly the same except for an "as ..." call,
2) They're each only called exactly once.

So I think I'd be inclined to save 20 lines and repetition and just put the "as ..." at each call-site.

Reply:
I think that's a good point - I do have to do a type conversion anyways, so I might as well do it in the callers.